File: dotnet-adapters-shortest-paths-example-project/main/roadrouting/CityRoadServiceHardcoded.cs
using System.Collections.Generic;

namespace roadrouting {
    /**
     * @author Tomas Johansson
     */
    public sealed class CityRoadServiceHardcoded : CityRoadService {

	    private IList<City> allCities;
	    private IList<Road> allRoads;
	    private City startCity = null;
	    private City endCity = null;
	
	    public CityRoadServiceHardcoded() {
		    SetHardcodedFieldsWithCitiesAndRoads();
	    }

	    public const string NAME_OF_START_CITY 	= "A";
	    public const string  NAME_OF_END_CITY 	= "F";

	    private void SetHardcodedFieldsWithCitiesAndRoads() {
		    startCity = CreateCity(NAME_OF_START_CITY); // "A"
		    City cityB = CreateCity("B");
		    City cityC = CreateCity("C");
		    City cityD = CreateCity("D");
		    City cityE = CreateCity("E");
		    endCity = CreateCity(NAME_OF_END_CITY); // "F"
		    allCities = new List<City>{startCity, cityB, cityC, cityD, cityE, endCity};

		    Road roadAB = CreateRoad(startCity, cityB, 100, RoadQuality.GOOD, "A to B");
		    Road roadAC = CreateRoad(startCity, cityC, 200, RoadQuality.BAD, "A to C");
		    Road roadAD = CreateRoad(startCity, cityD, 300, RoadQuality.GOOD, "A to D");
		    Road roadBC = CreateRoad(cityB, cityC, 150, RoadQuality.BAD, "B to C");
		    Road roadBD = CreateRoad(cityB, cityD, 250, RoadQuality.GOOD, "B to D");
		    Road roadBE = CreateRoad(cityB, cityE, 350, RoadQuality.BAD, "B to E");
		    Road roadCD = CreateRoad(cityC, cityD, 220, RoadQuality.GOOD, "C to D");
		    Road roadCE = CreateRoad(cityC, cityE, 340, RoadQuality.BAD, "C to E");
		    Road roadDE = CreateRoad(cityD, cityE, 130, RoadQuality.GOOD, "D to E");
		    Road roadDF = CreateRoad(cityD, endCity, 140, RoadQuality.GOOD, "D to F");
		    Road roadEF = CreateRoad(cityE, endCity, 150, RoadQuality.GOOD, "E to F");
		    allRoads = new List<Road>{roadAB, roadAC, roadAD, roadBC, roadBD, roadBE, roadCD, roadCE, roadDE, roadDF, roadEF};
	    }

	    private int cityKey = 1; // increased for each new City created 
	
	    private City CreateCity(string cityName) {
		    return new City(cityKey++, cityName);
	    }

	    private int roadKey = 1; // increased for each new Road created
	
	    private Road CreateRoad(
		    City cityFrom, 
		    City cityTo, 
		    int roadLength, 
		    RoadQuality roadQuality, 
		    string roadName
	    ) {
		    return new Road(roadKey++, cityFrom, cityTo, roadLength, roadQuality, roadName);
	    }

	    public IList<Road> GetAllRoads() {
		    return allRoads;
	    }
	
	    public IList<City> GetAllCities() {
		    return allCities;
	    }

	    public City GetStartCity() {
		    return startCity;
	    }
	    public City GetEndCity() {
		    return endCity;
	    }

	    public void ReleaseResourcesIfAny() {
		    // Nothing to do here in this class
	    }
    }
}

File: dotnet-adapters-shortest-paths-example-project/main/java/roadrouting/CityRoadServiceHardcoded.java
package roadrouting;

import java.util.Arrays;
import java.util.List;

/**
 * @author Tomas Johansson
 */
public final class CityRoadServiceHardcoded implements CityRoadService {

	private List<City> allCities;
	private List<Road> allRoads;
	private City startCity = null;
	private City endCity = null;
	
	public CityRoadServiceHardcoded() {
		setHardcodedFieldsWithCitiesAndRoads();
	}

	private void setHardcodedFieldsWithCitiesAndRoads() {
		startCity = createCity(NAME_OF_START_CITY); // "A"
		final City cityB = createCity("B");
		final City cityC = createCity("C");
		final City cityD = createCity("D");
		final City cityE = createCity("E");
		endCity = createCity(NAME_OF_END_CITY); // ("F"
		
		allCities = Arrays.asList(startCity, cityB, cityC, cityD, cityE, endCity);

		final Road roadAB = createRoad(startCity, cityB, 100, RoadQuality.GOOD, "A to B");
		final Road roadAC = createRoad(startCity, cityC, 200, RoadQuality.BAD, "A to C");
		final Road roadAD = createRoad(startCity, cityD, 300, RoadQuality.GOOD, "A to D");
		final Road roadBC = createRoad(cityB, cityC, 150, RoadQuality.BAD, "B to C");
		final Road roadBD = createRoad(cityB, cityD, 250, RoadQuality.GOOD, "B to D");
		final Road roadBE = createRoad(cityB, cityE, 350, RoadQuality.BAD, "B to E");
		final Road roadCD = createRoad(cityC, cityD, 220, RoadQuality.GOOD, "C to D");
		final Road roadCE = createRoad(cityC, cityE, 340, RoadQuality.BAD, "C to E");
		final Road roadDE = createRoad(cityD, cityE, 130, RoadQuality.GOOD, "D to E");
		final Road roadDF = createRoad(cityD, endCity, 140, RoadQuality.GOOD, "D to F");			
		final Road roadEF = createRoad(cityE, endCity, 150, RoadQuality.GOOD, "E to F");
		allRoads = Arrays.asList(roadAB, roadAC, roadAD, roadBC, roadBD, roadBE, roadCD, roadCE, roadDE, roadDF, roadEF);
		
	}

	private int cityKey = 1; // increased for each new City created 
	
	private City createCity(final String cityName) {
		return new City(cityKey++, cityName);
	}

	private int roadKey = 1; // increased for each new Road created
	
	private Road createRoad(
		final City cityFrom, 
		final City cityTo, 
		final int roadLength, 
		final RoadQuality roadQuality, 
		final String roadName
	) {
		return new Road(roadKey++, cityFrom, cityTo, roadLength, roadQuality, roadName);
	}

	public List<Road> getAllRoads() {
		return allRoads;
	}
	
	public List<City> getAllCities() {
		return allCities;
	}

	public City getStartCity() {
		return startCity;
	}
	public City getEndCity() {
		return endCity;
	}

	public void releaseResourcesIfAny() {
		// Nothing to do here in this class
	}
}